src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@SubJunk I don't think that's a good solution. As said above, I often rearrange/more code blocks, add exception handlers or otherwise insert code blocks which leads to indentation after I wrote the code itself. I also sometimes write the code using shorter identifier names and then change them to more descriptive names when I'm done. Having to actually change the code (from ternary to ```if...else``` as a result of what level of indentaton the code has at any given moment doesn't seem very practical or reasonable to me. 

There's also another case where I think ternaries is completely superior: If you make a complex method call where let's say you have 4 parameters the can be one thing or another that varies depending on circumstances. The handle that with ```if...else``` you either have to write 16 different method calls using with different parameters, or you have to use temporary variables that you set using ```if...else``` and then use the temporary variables in the call. Using ternaries in that situation not only saves a bunch of code, it also makes it much more readable in my opinion. It much easier when reading the code to see what leads to what because you don't have to "follow" the different temporary variables to figure out what the result is, or study all the different conditions to understand which method call is actually being performed. It allows you to "isolate" the evaluation for each parameter in an easily comprehensible way.

Thus I think we need to have a way to wrap ternary expressions when it's needed. I'm not saying we should use ternaries "all the time", I'm just saying that some times it's the best choice (this case isn't the best example) and when that's the case we need to agree how to do it.